src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -243,11 +243,27 @@ private void OnDeserialized(StreamingContext context)
         }
     }
 
+    [Flags]
+    internal enum StackTraceFormattingOptions
+    {
+        None                                 = 0,
+        ExcludeStackTraceHiddenAttribute     = 1 << 0,
+        ExcludeDispatchBoundaries            = 1 << 1,
+        ExcludeInnerExceptionBoundaries      = 1 << 2,
+        ResolveAsyncMethods                  = 1 << 3,
+        ResolveIteratorMethods               = 1 << 4,
+    }
 
     // Class which represents a description of a stack trace
     // There is no good reason for the methods of this class to be virtual.  
     public class StackTrace
     {
+        internal static StackTraceFormattingOptions FormattingOptions { get; set; }
```

Comment:
If this is the proposed opt-out mechanism, presumably it needs to be public

Reply:
Its `private public`, can switch it off with reflection 😉 

Will submit a corefx design for review when have agreed upon approach - though think we are close.